terraformignore.go

```diff
@@ -0,0 +1,219 @@
+package slug
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/scanner"
+)
+
+func parseIgnoreFile(rootPath string) []rule {
+	// Look for .terraformignore at our root path/src
+	file, err := os.Open(filepath.Join(rootPath, ".terraformignore"))
+	defer file.Close()
+
+	// If there's any kind of file error, punt and use the default ignore patterns
+	if err != nil {
+		// Only show the error debug if an error *other* than IsNotExist
+		if !os.IsNotExist(err) {
+			fmt.Printf("Error reading .terraformignore, default exclusions will apply: %v \n", err)
+		}
+		return defaultExclusions
+	}
+	return readRules(file)
+}
+
+func readRules(input io.Reader) []rule {
```

Comment:
Not saying we need to this at the moment, but this might be more efficient if the ignore rules are compiled into single state machine rather than a series of rules to iterate over for each path. It would really only be useful for very large numbers of files, which terraform configurations usually don't contain, but might be an easy win for those exceptional situations.

Reply:
Probably! In research I saw some implementations that stored rules in different data structures to make match-checking faster, so it's a fair point to make, but I think this is fast enough for now.